tests/core/base/test_tool.py

```diff
@@ -22,7 +22,7 @@ class FormatBook(BaseTool):
         def call(self) -> str:
             return f"{self.title} by {self.author}"
 
-    tool = FormatBook(title="The Name of the Wind", author="Patrick Rothfuss")
+    tool = FormatBook(title="The Name of the Wind", author="Patrick Rothfuss")  # pyright: ignore [reportCallIssue]
```

Comment:
aha, I think because you added `tool_call` to `BaseTool` it is now a required argument to create base tools when it wasn't before?

Reply:
I removed it